src/main/java/usi/si/seart/service/StaticCodeAnalysisService.java

```diff
@@ -0,0 +1,140 @@
+import java.util.stream.Collectors;
+
+/**
+ * Service responsible for performing static code analysis on git repositories
+ */
+
+@Slf4j
+@Service
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@AllArgsConstructor(onConstructor_ = @Autowired)
+public class StaticCodeAnalysisService {
+    //    ConversionService conversionService;
+
+    GitRepoClonerService gitRepoClonerService;
+
+    GitRepoRepository gitRepoRepository;
+
+
+    /**
+     * Computes the set of code metrics of a given repository.
+     *
+     * @param repo the git repo
+     * @param persist   whether to persist metrics for the given repository.
+     *                  If true, a repo with that repo_name needs to exist in the DB.
+     *                  If false, the GitRepo object won't be fetched.
+     * @return the set of code metrics
+     * @throws StaticCodeAnalysisException if an error occurred while performing static code analysis.
+     */
+    @Async("GitCloning")
+    public Future<HashSet<GitRepoMetric>> getCodeMetrics(@NotNull GitRepo repo, boolean persist) throws StaticCodeAnalysisException {
+        Gson g = new Gson();
+        HashSet<GitRepoMetric> metrics;
+        // Deletes the temporary folder of the repo once outside of this clause.
+        try (ClonedRepo clonedRepo = gitRepoClonerService.cloneRepo(new URL("https://github.com/" + repo.getName())).get()) {
+            // Runs cloc for gathering code metrics
+            String output = new TerminalExecution(clonedRepo.getPath(), "cloc --json --quiet .").start().getStdout().lines().collect(Collectors.joining("\n"));
+
+            // TODO: I think this can lead to side effects
+//            metrics = conversionService.convert(Pair.of(repo,g.fromJson(output, JsonObject.class)), HashSet.class);
+//            metrics = (new JsonObjectToRepoMetrics()).convert(g.fromJson(output, JsonObject.class)));
+            if (!persist || repo == null) {
+                return new AsyncResult<>(convert(null, g.fromJson(output, JsonObject.class)));
+            }
+
+            // Converts the string output from 'cloc' into a set of code metrics.
+            metrics = convert(repo, g.fromJson(output, JsonObject.class));
+            repo.setMetrics(metrics);
+            repo.setCloned();
+            gitRepoRepository.save(repo);
+            log.info("Stored code metrics for repository '"+repo.getName()+"'");
+
+
+        } catch (InterruptedException | ExecutionException | TerminalExecutionException | MalformedURLException e) {
+            log.error("Could not compute code metrics for repository '"+repo.getName()+"'", e);
+            throw new StaticCodeAnalysisException(e);
+        }
+
+        return new AsyncResult<>(metrics);
+    }
+    /**
+     * Computes the set of code metrics of a given repository.
+     *
+     * @param repo_name the name of the git repository (eg "/AlbertCerfeda/example" )
+     * @see StaticCodeAnalysisService::getCodeMetrics(GitRepo, boolean)
```

Comment:
I think the correct JDoc syntax is `@see StaticCodeAnalysisService#getCodeMetrics(GitRepo, boolean)`